core/src/main/java/com/datastax/oss/driver/api/core/cql/ColumnDefinitions.java

```diff
@@ -97,22 +100,59 @@ default ColumnDefinition get(@NonNull CqlIdentifier name) {
   /** Whether there is a definition using the given CQL identifier. */
   boolean contains(@NonNull CqlIdentifier id);
 
+  /**
+   * Returns the indices of all columns that use the given name.
+   *
+   * <p>Because raw strings are ambiguous with regard to case-sensitivity, the argument will be
+   * interpreted according to the rules described in {@link AccessibleByName}.
+   *
+   * @return the indices, or an empty list if no column uses this name.
+   * @apiNote the default implementation only exists for backward compatibility. It wraps the result
+   *     of {@link #firstIndexOf(String)} in a singleton list, which is not entirely correct, as it
+   *     will only return the first occurrence. Therefore it also logs a warning.
+   *     <p>Implementors should always override this method (all built-in driver implementations
+   *     do).
+   */
+  @NonNull
+  default List<Integer> allIndicesOf(@NonNull String name) {
+    Loggers.COLUMN_DEFINITIONS.warn(
+        "Implementors should always override this method, "
+            + "the default implementation only returns the first occurrence");
+    return Collections.singletonList(firstIndexOf(name));
```

Comment:
I wouldn't use this as a general solution to breaking API changes, but a number of things make me think it's OK for this case:
* this only matters for prepared and bound statements (UDTs can't have duplicate field names, and rows are not settable).
* there are probably very few (if any) custom implementations of `PreparedStatement` and `BoundStatement` outside of the driver.
* even so, the implementation is only incorrect when there are multiple occurrences of the same variable, which I still think is an edge case.
* finally, the default implementation is not completely broken, it just functions like previous 4.x releases.